pulpcore/tasking/_util.py

```diff
@@ -97,7 +97,26 @@ def perform_task(task_pk, task_working_dir_rel_path):
     connection.connection = None
     # enc_args and enc_kwargs are deferred by default but we actually want them
     task = Task.objects.defer(None).select_related("pulp_domain").get(pk=task_pk)
-    user = get_users_with_perms(task, with_group_users=False).first()
+    # These queries were specifically constructed and ordered this way to ensure we have the highest
+    # chance of getting the user who dispatched the task since we don't have a user relation on the
+    # task model. The second query acts as a fallback to provide ZDU support. Future changes will
+    # require to keep these around till a non-ZDU release is planned (probably 3.70 the earliest).
```

Comment:
Technically, if we get it right, the breaking change releases still allow upgrades without downtime.
```suggestion
    # require to keep these around till a breaking change release is planned (probably 3.70 the earliest).
```